### ISSUE TITLE
Bug: Page types search returns multiple entries when asked for exact value

### ISSUE THREAD

**Issue by andrzejewsky** [saleor/saleor#13249]
### What are you trying to achieve?

The search for productTypes query doesn't work properly. It returns multiple elements even if we use exact name. That cause weird UX experience Ref: https://github.com/saleor/saleor-dashboard/issues/3804

### Steps to reproduce the problem

1. Create a few similar entries (eg. content-page-1,  content-page-2,  content-page-3)
```graphql
mutation {
  pageTypeCreate(input: { name:"content-page-N" }) {
		pageType {
     name
    }
  }
}
```
3. Try to search, using exact name of one of these
```graphql
{
  pageTypes(filter:{ search:"content-page-1" }, first: 10) {
    edges {
      node {
        name
      }
    }
  }
```
4. Api returns all of content-page-N elements

### What did you expect to happen?

The Api should return only one element when I use its exact name

### Environment

Saleor version: 3.14

**Comment by andrzejewsky** [comment]
Update: bug is related to any search in the API, where we search by using some query text. It returns no response or "similar" entries.

**Comment by aniav** [comment]
I don't think this is really an issue, search will return exact and similar results by definition. And filtering a specific field will return exact matches.

Or am I missing anything her?
